src/modes/ModeEstimation.jl

```diff
@@ -17,6 +17,8 @@ import Printf
 struct MLE end
 struct MAP end
 
+struct constrained end
+struct unconstrained end
```

Comment:
Maybe rather make it a type parameter:
```suggestion
struct MLE{constrained} end
struct MAP{constrained} end

MLE() = MLE{false}()
MAP() = MAP{false}()
```
BTW I'm not sure if the default should actually be to optimize in the constrained space (so that models don't have to deal with linking etc.).

Reply:
For now we should set the default to the unconstrained space, and then we can change it to use the constrained space when it's a little easier to get bounds.

Reply:
That sounds sensible. Will work that in.